contracts/strategies/savvy/savvy-adapter.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+
+import "../../interfaces/savvy.sol";
+import "../../lib/TokenUtils.sol";
+import "../../lib/Checker.sol";
+import "../../interfaces/vault.sol";
+import "../../base/Errors.sol";
+
+
+contract ContraxSavvyAdapter is ITokenAdapter, Initializable, Ownable2StepUpgradeable {
```

Comment:
Since this contract is upgradeable you'll need to disable initializer in the constructor 

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/utils/Initializable.sol#L39-L54

https://github.com/savvy-finance/savvy-contracts/blob/main/contracts/adapters/beefy/BeefyTokenAdapter.sol#L32-L34